app-backend/batch/src/main/scala/com/azavea/rf/batch/landsat8/ImportLandsat8.scala

```diff
@@ -0,0 +1,274 @@
+package com.azavea.rf.batch.landsat8
+
+import com.azavea.rf.batch.Job
+import com.azavea.rf.batch.util.S3
+import com.azavea.rf.database.{Database => DB}
+import com.azavea.rf.database.tables._
+import com.azavea.rf.datamodel._
+
+import io.circe._
+import io.circe.syntax._
+import geotrellis.proj4.CRS
+import geotrellis.slick.Projected
+import geotrellis.vector._
+import jp.ne.opt.chronoscala.Imports._
+
+import java.time.{LocalDate, ZoneOffset}
+import java.util.UUID
+
+import scala.io.Source
+import scala.collection.mutable.ListBuffer
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.duration.Duration
+import scala.concurrent.{Await, Future}
+import scala.util.{Failure, Success}
+import scala.util.control.Breaks._
+
+case class ImportLandsat8(startDate: LocalDate = LocalDate.now(ZoneOffset.UTC), threshold: Int = 10)(implicit val database: DB) extends Job {
+  val name = "import_landsat8"
+
+  @SuppressWarnings(Array("OptionGet"))
+  protected def getCsv: ListBuffer[Map[String, String]] = {
+    val source = Source.fromURL(landsat8Config.usgsLandsatUrl, "UTF-8")
+    val lines = source.getLines()
+
+    val endDate = startDate + 1.day
+    val buffer = ListBuffer[Map[String, String]]()
+    val idToIndex: Map[String, Int] = lines.next().split(",").map(_.trim).zipWithIndex.toMap
+    val indexToId = idToIndex map { case (v, i) => i -> v }
+
+    var counter = 0
+    breakable {
+      lines foreach { line =>
+        val row =
+          line
+            .split(",")
+            .map(_.trim)
+            .zipWithIndex
+            .map { case (v, i) => indexToId(i) -> v }
+            .toMap
+
+        row.get("acquisitionDate") foreach { dateStr =>
+          val date = LocalDate.parse(dateStr)
+          if (startDate <= date && endDate > date) buffer += row
+          else if (date < startDate) counter += 1
+        }
+
+        if (counter > threshold) break
+      }
+    }
+
+    buffer
+  }
+
+  protected def getLandsatPath(landsatId: String) = {
+    val (wPath, wRow) = landsatId.substring(3, 6) -> landsatId.substring(6, 9)
+    val path = s"L8/$wPath/$wRow/$landsatId"
+    logger.debug(s"Constructed path: $path")
+    path
+  }
+
+  protected def getLandsatUrl(landsatId: String) = {
+    val rootUrl = s"https://landsat-pds.s3.amazonaws.com/${getLandsatPath(landsatId)}"
+    logger.debug(s"Constructed Root URL: $rootUrl")
+    rootUrl
+  }
+
+  protected def s3ObjExists(url: String): Boolean = {
+    try {
+      Source.fromURL(url, "UTF-8")
+      true
+    } catch {
+      case _: java.io.FileNotFoundException => false
+    }
+  }
+
+  protected def sizeFromPath(tifPath: String, landsatId: String): Int = {
+    val path = s"${getLandsatPath(landsatId)}$tifPath"
+    logger.info(s"Getting object size for path: $path")
+    S3.getObject(landsat8Config.bucketName, path).getObjectMetadata.getContentLength.toInt
```

Comment:
Getting an exception here -- I think because the bucket is in `us-west-1`:

```
06:34:27.808 [INFO ] c.a.r.b.l.ImportLandsat8 - Getting object size for path: L8/096/016/LC80960162016080LGN00/LC80960162016080LGN00_B8.TIF
Exception in thread "main" com.amazonaws.services.s3.model.AmazonS3Exception: The authorization header is malformed; the region 'us-east-1' is wrong; expecting 'us-west-2' (Service: Amazon S3; Status
 Code: 400; Error Code: AuthorizationHeaderMalformed; Request ID: FF67B848CC16BED9), S3 Extended Request ID: kJWCE76K0/S34N91a/9oPQECmgGe17T55TsYGe85YKQocntmyVYUHEzQ4Nmlw6OBX0vedrWlCdA=
        at com.amazonaws.http.AmazonHttpClient$RequestExecutor.handleErrorResponse(AmazonHttpClient.java:1579)
        at com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeOneRequest(AmazonHttpClient.java:1249)
        at com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeHelper(AmazonHttpClient.java:1030)
        at com.amazonaws.http.AmazonHttpClient$RequestExecutor.doExecute(AmazonHttpClient.java:742)
        at com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeWithTimer(AmazonHttpClient.java:716)
        at com.amazonaws.http.AmazonHttpClient$RequestExecutor.execute(AmazonHttpClient.java:699)
        at com.amazonaws.http.AmazonHttpClient$RequestExecutor.access$500(AmazonHttpClient.java:667)
        at com.amazonaws.http.AmazonHttpClient$RequestExecutionBuilderImpl.execute(AmazonHttpClient.java:649)
        at com.amazonaws.http.AmazonHttpClient.execute(AmazonHttpClient.java:513)
        at com.amazonaws.services.s3.AmazonS3Client.invoke(AmazonS3Client.java:4194)
        at com.amazonaws.services.s3.AmazonS3Client.invoke(AmazonS3Client.java:4141)
        at com.amazonaws.services.s3.AmazonS3Client.getObject(AmazonS3Client.java:1384)
        at com.amazonaws.services.s3.AmazonS3Client.getObject(AmazonS3Client.java:1262)
        at com.azavea.rf.batch.util.S3$.getObject(S3.scala:21)
        at com.azavea.rf.batch.landsat8.ImportLandsat8.sizeFromPath(ImportLandsat8.scala:89)
        at com.azavea.rf.batch.landsat8.ImportLandsat8$$anonfun$csvRowToScene$1$$anonfun$14.apply(ImportLandsat8.scala:183)
        at com.azavea.rf.batch.landsat8.ImportLandsat8$$anonfun$csvRowToScene$1$$anonfun$14.apply(ImportLandsat8.scala:180)
        at scala.collection.immutable.List.map(List.scala:273)
        at com.azavea.rf.batch.landsat8.ImportLandsat8$$anonfun$csvRowToScene$1.apply(ImportLandsat8.scala:180)
        at com.azavea.rf.batch.landsat8.ImportLandsat8$$anonfun$csvRowToScene$1.apply(ImportLandsat8.scala:118)
        at scala.concurrent.impl.Future$PromiseCompletingRunnable.liftedTree1$1(Future.scala:24)
        at scala.concurrent.impl.Future$PromiseCompletingRunnable.run(Future.scala:24)
        at scala.concurrent.impl.ExecutionContextImpl$AdaptedForkJoinTask.exec(ExecutionContextImpl.scala:121)
        at scala.concurrent.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260)
        at scala.concurrent.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339)
        at scala.concurrent.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979)
        at scala.concurrent.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107)
```